src/main/java/de/tum/in/www1/artemis/web/rest/TextExerciseResource.java

```diff
@@ -199,4 +215,71 @@ public TextExerciseResource(TextExerciseRepository textExerciseRepository, UserS
+        if (participation == null) {
+            return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert(ENTITY_NAME, "participationNotFound", "No participation was found for the given ID.")).body(null);
+        }
+        TextExercise textExercise;
+        if (participation.getExercise() instanceof TextExercise) {
+            textExercise = (TextExercise) participation.getExercise();
+            if (textExercise == null) {
+                return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert("textExercise", "exerciseEmpty", "The exercise belonging to the participation is null.")).body(null);
+            }
+        } else {
+            return ResponseEntity.badRequest().headers(HeaderUtil.createFailureAlert("textExercise", "wrongExerciseType", "The exercise of the participation is not a modeling exercise.")).body(null);
+        }
+
+        // users can only see their own submission (to prevent cheating), TAs, instructors and admins can see all models
+        if (!authCheckService.isOwnerOfParticipation(participation) && !courseService.userHasAtLeastTAPermissions(textExercise.getCourse())) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
+
+
+        // if no results, check if there are really no results or the relation to results was not updated yet
+        if (participation.getResults().size() <= 0) {
+            List<Result> results = resultRepository.findByParticipationIdOrderByCompletionDateDesc(participation.getId());
+            participation.setResults(new HashSet<>(results));
+        }
+
+        ObjectNode data = objectMapper.createObjectNode();
+
+        TextSubmission textSubmission = participation.findLatestTextSubmission();
+        // NOTE: avoid infinite recursion by setting submissions to null
+        participation.setSubmissions(null);
+        data.set("participation", objectMapper.valueToTree(participation));
+        if (textSubmission != null) {
+            // set reference to participation if null
+            if (textSubmission.getParticipation() == null) {
+                textSubmission.setParticipation(participation);
+            }
+//            Result result = textSubmission.getResult();
+//            if (textSubmission.isSubmitted() && result != null && result.isRated()) {
+//                // find assessments if textSubmission is submitted and result is rated
+//                String assessment = textAssessmentService.findLatestAssessment(textExercise.getId(), participation.getStudent().getId(), textSubmission.getId());
+//                if (assessment != null && assessment != "") {
+//                    try {
+//                        data.set("assessments", objectMapper.readTree(assessment));
+//                    } catch (IOException e) {
+//                        log.error("Error while reading assessment JSON: {}", e.getMessage());
+//                    }
+//                }
+//            }
```

Comment:
What are your plans with this piece of code?

Reply:
This is a placeholder, just to remember this is something that has to be implemented. I don't know if you are already implementing this

Reply:
Please add a TODO to this code then, otherwise it might get los